Display real cpu time in explain analyze

arhimondr opened the pull request:
```
presto:default> explain analyze select linestatus, count(orderkey) from lineitem group by 1;
                                                                                Query Plan                                                                                 
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Fragment 1 [HASH]                                                                                                                                                         
     CPU: 31.86ms, Scheduled: 140.80ms, Input: 10 rows (240B); per task: avg.: 10.00 std.dev.: 0.00, Output: 2 rows (30B)                                                  
     Output layout: [linestatus, count]                                                                                                                                    
     Output partitioning: SINGLE []                                                                                                                                        
     Grouped Execution: false                                                                                                                                              
     - Project[] => [linestatus:varchar(1), count:bigint]                                                                                                                  
             CPU: 2.00ms (0.10%), Scheduled: 5.00ms (0.04%), Input: 2 rows (48B), Output: 2 rows (30B), Filtered: 0.00%                                                    
             Input avg.: 0.13 rows, Input std.dev.: 264.58%                                                                                                                
         - Aggregate(FINAL)[linestatus][$hashvalue] => [linestatus:varchar(1), $hashvalue:bigint, count:bigint]                                                            
                 CPU: 4.00ms (0.20%), Scheduled: 7.00ms (0.06%), Output: 2 rows (48B)                                                                                      
                 Input avg.: 0.63 rows, Input std.dev.: 264.58%                                                                                                            
                 count := "count"("count_9")                                                                                                                               
             - LocalExchange[HASH][$hashvalue] ("linestatus") => linestatus:varchar(1), count_9:bigint, $hashvalue:bigint                                                  
                     CPU: 5.00ms (0.24%), Scheduled: 23.00ms (0.18%), Output: 10 rows (240B)                                                                               
                     Input avg.: 0.63 rows, Input std.dev.: 186.55%                                                                                                        
                 - RemoteSource[2] => [linestatus:varchar(1), count_9:bigint, $hashvalue_10:bigint]                                                                        
                         CPU: 1.00ms (0.05%), Scheduled: 5.00ms (0.04%), Output: 10 rows (240B)                                                                            
                         Input avg.: 0.63 rows, Input std.dev.: 186.55%                                                                                                    
                                                                                                                                                                           
 Fragment 2 [SOURCE]                                                                                                                                                       
     CPU: 2.03s, Scheduled: 7.08s, Input: 6001215 rows (137.36MB); per task: avg.: 6001215.00 std.dev.: 0.00, Output: 10 rows (240B)                                       
     Output layout: [linestatus, count_9, $hashvalue_11]                                                                                                                   
     Output partitioning: HASH [linestatus][$hashvalue_11]                                                                                                                 
     Grouped Execution: false                                                                                                                                              
     - Aggregate(PARTIAL)[linestatus][$hashvalue_11] => [linestatus:varchar(1), $hashvalue_11:bigint, count_9:bigint]                                                      
             CPU: 392.00ms (19.18%), Scheduled: 436.00ms (3.47%), Output: 10 rows (240B)                                                                                   
             Input avg.: 1200243.00 rows, Input std.dev.: 11.99%                                                                                                           
             count_9 := "count"("orderkey")                                                                                                                                
         - ScanProject[table = hive:default:lineitem, grouped = false] => [orderkey:bigint, linestatus:varchar(1), $hashvalue_11:bigint]                                   
                 Cost: {rows: 6001215 (85.85MB), cpu: 90018225.00, memory: 0.00, network: 0.00}/{rows: 6001215 (137.36MB), cpu: 234047385.00, memory: 0.00, network: 0.00} 
                 CPU: 1.64s (80.23%), Scheduled: 12.10s (96.22%), Input: 6001215 rows (215.05MB), Output: 6001215 rows (137.36MB), Filtered: 0.00%                         
                 Input avg.: 1200243.00 rows, Input std.dev.: 11.99%                                                                                                       
                 $hashvalue_11 := "combine_hash"(bigint '0', COALESCE("$operator$hash_code"("linestatus"), 0))                                                             
                 LAYOUT: default.lineitem                                                                                                                                  
                 orderkey := orderkey:bigint:0:REGULAR                                                                                                                     
                 linestatus := linestatus:varchar(1):9:REGULAR                                                                                                             
                                                                                                                                                                           
                                                                                                                                                                           
(1 row)
```

martint commented on presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanNodeStatsSummarizer.java:
```diff
@@ -89,8 +89,8 @@ private PlanNodeStatsSummarizer() {}
 
-                long wall = operatorStats.getAddInputWall().toMillis() + operatorStats.getGetOutputWall().toMillis() + operatorStats.getFinishWall().toMillis();
-                planNodeWallMillis.merge(planNodeId, wall, Long::sum);
+                long cpu = operatorStats.getAddInputCpu().toMillis() + operatorStats.getGetOutputCpu().toMillis() + operatorStats.getFinishCpu().toMillis();
```
Instead of replacing wall time with cpu time, we should show both.

arhimondr commented:
@martint updated

arhimondr commented:
@dain 

> Now we have real timings for everything, so we could show real values everywhere

So many times i ended up using the calculator to figure out how much CPU time. Is the percentage of a single operation in a stage is any useful? If it was a percentage of the entire query - than yes. But the usefulness of the fraction of a stage is very questionable. 

Let's display the CPU time as duration, and change the fraction to reflect the CPU fraction in terms of the entire query.

CC: @raghavsethi @martint

arhimondr commented:
@dain I added absolute CPU/Scheduled timings per operator. Also i changed fraction to be relative to the overall CPU/Scheduled timings.

CC: @raghavsethi @martint

dain commented:
@arhimondr can you add a sample to the PR description?

martint commented on presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanNodeStats.java:
```diff
@@ -38,7 +38,8 @@
 
-    private final Duration planNodeWallTime;
+    private final Duration planNodeCpuTime;
+    private final Duration planNodeScheduledTime;
```
I agree. Regardless, the rename should happen in a separate change. As it stands now, it looks as if we removed something (wall time) and added something else (scheduled time), even though this change just adds CPU time.

arhimondr replied:
Btw, I'm renaming it to scheduled because it is called `totalScheduledTime` in `StageStats`

martint commented on presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java:
```diff
@@ -354,7 +354,13 @@ private void printStats(int indent, PlanNodeId planNodeId, boolean printInput, b
-        output.append("CPU fraction: " + formatDouble(cpuTimeFraction) + "%, Scheduled fraction: " + formatDouble(scheduledTimeFraction) + "%");
+        output.append(format(
+                "CPU: %s (%s%%), Scheduled: %s (%s%%)",
+                nodeStats.getPlanNodeCpuTime(),
```
You may need a `.convertToMostSuccinctTimeUnit()` here and below.
